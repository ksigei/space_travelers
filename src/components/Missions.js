import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import getMissionsAction from '../redux/missions/actions';

const Missions = () => {
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(getMissionsAction());
  }, []);
  const missions = useSelector((state) => state.missionReducers);
  return missions.map((mission) => (
    <main className="box" key={mission.missionId}>
      <div className="left">
        {/* <div className="mission-img">
          <img src={mission.missionImgUrl} alt="mission" />
        </div> */}
      </div>
      <div className="right">
        <h4>{mission.missionName}</h4>
        <p>{mission.missionDescription}</p>
        <button
          type="button"
          className="btn btn-primary btn-md"
        >
          Join mission

        </button>
      </div>
    </main>
  ));
};

export default Missions;

