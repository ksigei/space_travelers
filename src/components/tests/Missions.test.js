import React from 'react';
import renderer from 'react-test-renderer';
import '../Missions';

test('matches snapshot', () => {
  const tree = renderer.create(
    <tr>
      <th>Mission</th>
      <th>Description</th>
      <th>Status</th>
    </tr>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
