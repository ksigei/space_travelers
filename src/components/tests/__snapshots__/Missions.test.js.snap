// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<tr>
  <th>
    Mission
  </th>
  <th>
    Description
  </th>
  <th>
    Status
  </th>
</tr>
`;
